Ratelimit presence updates (#18000)
element-hq/synapse: docs/usage/configuration/config_documentation.md

  concurrent: 5
```
---
### `rc_set_presence`

This option sets ratelimiting for presence.

> The `rc_set_presence.per_user` sets ratelimiting how often a specific users' presence
> updates are evaluated. Ratelimited presence updates are ignored.
> `per_user` defaults to `per_second: 0.1`, `burst_count: 1`.

Reviewer: Slight clarification:

```suggestion
The `rc_set_presence.per_user` option sets rate limits on how often a specific
users' presence updates are evaluated. Ratelimited presence updates are
ignored, and no error is returned to the client.

`per_user` defaults to `per_second: 0.1`, `burst_count: 1`.
```

Example configuration:
```yaml
rc_set_presence:
  per_user:

>     per_second: 0.1
>     burst_count: 1

Reviewer: Typically the example configuration should be something other than the default.

```suggestion
    per_second: 0.05
    burst_count: 0.5
```

Reviewer 2: `burst_count` of 0.5 will be converted to an integer (0), which effectively ignores _all_ presence updates. Maybe a better example should be provided?

Reviewer: Thanks for noticing! Fixed in https://github.com/element-hq/synapse/pull/18159.

  concurrent: 5
```
---
### `rc_presence`

This option sets ratelimiting for presence.

The `rc_presence.per_user` option sets rate limits on how often a specific
users' presence updates are evaluated. Ratelimited presence updates sent via sync are
ignored, and no error is returned to the client.

> This option also sets the rate limits used in the `/_matrix/client/v3/presence` endpoint.

Reviewer: ```suggestion
This option also sets the rate limit for the
[`PUT /_matrix/client/v3/presence/{userId}/status`](https://spec.matrix.org/latest/client-server-api/#put_matrixclientv3presenceuseridstatus)
endpoint.
```
